TopMovies/MoviesConnector/CategoryVCConnector.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  CategoryVCConnector.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 17.02.2021.
+//
+
+import Foundation
+
+class CategoryVCConnector: BaseConnector<MoviesCategoryVCProps> {
+  var categoryId: MovieCategory.ID?
+  var downloadedMovies: [Movie.ID] = []
+  var isExistingMoviesConnected = false
+  
+  required init(updateProps: @escaping (MoviesCategoryVCProps) -> Void) {
+    super.init(updateProps: updateProps)
+  }
+  convenience init(categoryId: MovieCategory.ID, updateProps: @escaping (MoviesCategoryVCProps) -> Void) {
+    self.init(updateProps: updateProps)
+    self.categoryId = categoryId
+    mainStore.dispatch(RequestedMoviesListAction(categoryId: categoryId, requestType: .loadMore))
+  }
+  
+  override func newState(state: MainState) {
```

Comment:
That function became really large and it's hard to understand what's exactly going on here.